minvws/nl-kat-coordination
# Update a Boefje

**NOTE:** This PR is a follow-up of PR #3456 

------------------------

### Changes
This PR makes it possible to update a Boefje variant.

#### Features added
**Boefje detail page:**
- "Edit variant" button on Boefje detail page (in variants table)
- Variants table: add "Boefje ID"
- Variants table: add age column
- Variants table: fix created date (make it readable)

**New Boefje form page:**
- Change submit/discard button names to "Create new Boefje" and "Discard new Boefje"

**Edit Boefje form page:**
- "Edit Boefje" form page that is pre-filled
- Update helpertext for "Container image" input field
- Return to updated Boefje page

**Boefje variant form page:**
- Fix breadcrumbs
- Remove read-only from "Container image" input field


**Code:**
- In View: improve code and reduce duplicate code

### Issue link
Closes #3468

### Demo
![afbeelding](https://github.com/user-attachments/assets/5aac4254-bdd8-471e-8c41-228041beb660)
![afbeelding](https://github.com/user-attachments/assets/1fe3cf8d-687a-4f98-9f61-0457a11672a2)



### QA notes
First create a variant of Nmap TCP:
- Go to "Nmap TCP" boefje
- Check the "Variants" table. There should be one item (the current Boefje)
- Click "Add variant"
- Type a different name, make some small changes to the other input fields
- Click "Create variant"
- You should return to the "Nmap TCP" boefje

Then, edit the variant:
- Check the "Variants" table. Unfold your just created variant.
- Click "Edit variant"
- Check if **all** fields in de form are pre-filled
- Type a different name, make some small changes to the other input fields
- Click "Save changes"
- You should return to the variant boefje detail page
- Check if the changes were saved

---

### Code Checklist







---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
